src/client/Microsoft.Identity.Client/ManagedIdentity/ManagedIdentityClient.cs

```diff
@@ -157,5 +162,24 @@ private static bool ValidateAzureArcEnvironment(string identityEndpoint, string
             logger?.Verbose(() => "[Managed Identity] Azure Arc managed identity is not available.");
             return false;
         }
+
+        /// <summary>
+        /// Sets the in-memory binding certificate used to prime the mtls_pop scheme on subsequent requests.
+        /// </summary>
+        /// <remarks>
+        /// Disposing an <see cref="X509Certificate2"/> releases resources for this in-memory instance;
+        /// it does <b>not</b> remove a certificate from any OS certificate store (store removal requires <see cref="X509Store.Remove(X509Certificate2)"/>).
+        /// </remarks>
+        internal void SetRuntimeMtlsBindingCertificate(X509Certificate2 cert)
+        {
+            // Atomically swap the reference and dispose the previous one (if different).
+            var old = System.Threading.Interlocked.Exchange(ref _runtimeMtlsBindingCertificate, cert);
+
+            // If the same instance is passed again, do not dispose it (it is now the current value).
+            if (!object.ReferenceEquals(old, cert))
+            {
+                old?.Dispose();
```

Comment:
Pls don't explicitly call dispose. What if you return the certificate to the caller (via AuthenticationResult) and they use it for calling downstream APIs but you dipose of it explicitly.